cassandraprotocol/frame/decoder.go

```diff
@@ -45,27 +56,29 @@ func (r RawHeader) String() string {
 // one must either call DecodeBody or DiscardBody to fully read or discard the body contents.
 func (c *Codec) DecodeHeader(source io.Reader) (*RawHeader, error) {
 	if versionAndDirection, err := primitives.ReadByte(source); err != nil {
-		return nil, fmt.Errorf("cannot decode header version and direction: %w", err)
+		return nil, err
```

Comment:
Why change this?

Reply:
I'll answer all of the comments related to this here.
The underlying `err` is not being exposed in a way that the user can access it. This is wrapping an error by embedding it in a string which makes it pretty much impossible for any error handling at a higher level. I started integrating this by passing a `connection` object but then I would lose all error handling that I already had which would handle `io` errors.

Maybe we can agree to revert this to `fmt.Errorf` but we will have to use a temporary buffer to read from the connection, handle connection errors and only then pass that buffer to `cassandraprotocol`

Reply:
I had no idea that this was actually wrapping errors and the user can unwrap them, TIL!

Reply:
done